[Docs][PyOV] Documentation for Custom Python Operation (#25615)
openvinotoolkit/openvino: docs/articles_en/documentation/openvino-extensibility.rst

         :language: cpp
         :fragment: [add_frontend_extension]

When Python API is used, there is no way to implement a custom OpenVINO operation. Even if custom OpenVINO operation is implemented in C++ and loaded into the runtime by a shared library, there is still no way to add a frontend mapping extension that refers to this custom operation. In this case, use C++ shared library approach to implement both operations semantics and framework mapping.

> 

Reviewer: We lost that part about the inability to add mapping extension for ops loaded from a library if they were not added beforehand when op was implemented. That part is not fixed in the recent version of API, right?

Author: returned back

         :language: cpp
         :fragment: [add_frontend_extension]

> If custom OpenVINO operation is implemented in C++ and loaded into the runtime using Python API by a shared library, there is no way to add a frontend mapping extension that refers to this custom operation. In this case, use C++ shared library approach to implement both operations semantics and framework mapping.

Reviewer: This limitation exists for both C++ and Python.

Author: updated

         :language: cpp
         :fragment: [add_frontend_extension]

The Python API now supports the implementation of custom OpenVINO operations, allowing for direct integration within Python code. Even if custom OpenVINO operation is implemented in C++ and loaded into the runtime by a shared library, there is still no way to add a frontend mapping extension that refers to this custom operation. In this case, use C++ shared library approach to implement both operations semantics and framework mapping.

> Python can still be used to map and decompose operations when only operations from the standard OpenVINO operation set are used.
> 

Author: ```suggestion
Python can still be used to map and decompose operations when operations from the standard OpenVINO operation set and custom operation implemented in Python are used.

```

> The Python API now supports the implementation of custom OpenVINO operations, allowing for direct integration within Python code. If a custom operation is implemented in C++ and loaded as a shared library, Python can also reference this operation through frontend mapping extensions. This provides flexibility for developers to implement custom operations in either Python or C++. Even if custom OpenVINO operation is implemented in C++ and loaded into the runtime by a shared library, there is still no way to add a frontend mapping extension that refers to this custom operation. In this case, use C++ shared library approach to implement both operations semantics and framework mapping.

Author: ```suggestion
The Python API now supports the implementation of custom OpenVINO operations, allowing for direct integration within Python code. Even if custom OpenVINO operation is implemented in C++ and loaded into the runtime by a shared library, there is still no way to add a frontend mapping extension that refers to this custom operation. In this case, use C++ shared library approach to implement both operations semantics and framework mapping.
```

         :language: cpp
         :fragment: [add_frontend_extension]

> The Python API now supports implementation of custom OpenVINO operations, enabling direct integration within Python code. Still python custom operations won't be loaded into the runtime, so use the C++ shared library approach to implement both operation semantics and framework mapping that refers to this custom operation.

Reviewer: This is not a release notes, I suppose. So `now` is redundant.

Reviewer: The meaning of the whole sentence doesn't look correct: "Still python custom operations won't be loaded into the runtime, so use the C++ shared library approach to implement both operation semantics and framework mapping that refers to this custom operation.". By definition of Python custom operation you already have "loaded" it in runtime - just insert it into the graph and it will work. Am I missing something?

Author: looks like it really makes no sense anymore. Deleted.

> The Python API now supports implementation of custom OpenVINO operations, enabling direct integration within Python code. Even if a custom OpenVINO operation is implemented in C++ and loaded into the runtime by a shared library, there is still no way to add a frontend mapping extension that refers to this custom operation. In this case, use the C++ shared library approach to implement both operation semantics and framework mapping.

Author: ```suggestion
The Python API now supports implementation of custom OpenVINO operations, enabling direct integration within Python code. Still python custom operations won't be loaded into the runtime, so use the C++ shared library approach to implement both operation semantics and framework mapping.
```

> Python can still be used to map and decompose operations when operations from the standard OpenVINO operation set and custom operation implemented in Python are used.

Reviewer: But only via `ConversionExtension` in case of Python, right?

Author: yes
